### PR TITLE
[Merged by Bors] - feat(Geometry/RingedSpace/OpenImmersion): make `IsOpenImmersion` instance

### PR DESCRIPTION
**Pull request by jjaassoonn**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Geometry/RingedSpace/OpenImmersion.lean [hunk 1]
```diff
@@ -129,7 +129,7 @@ noncomputable def isoRestrict : X ≅ Y.restrict H.base_open :=
 #align algebraic_geometry.PresheafedSpace.is_open_immersion.iso_restrict AlgebraicGeometry.PresheafedSpace.IsOpenImmersion.isoRestrict
 
 @[simp]
-theorem isoRestrict_hom_ofRestrict : H.isoRestrict.hom ≫ Y.ofRestrict _ = f := by
+theorem isoRestrict_hom_ofRestrict : (isoRestrict f).hom ≫ Y.ofRestrict _ = f := by
```
Comment (jjaassoonn) [inline Mathlib/Geometry/RingedSpace/OpenImmersion.lean:132]: Maybe we should change the namespace to enable dot notation as well? @erdOne what do you think?

Comment (jjaassoonn) [reply Mathlib/Geometry/RingedSpace/OpenImmersion.lean:132]: It seems
```lean
noncomputable def _root_.AlgebraicGeometry.PresheafedSpace.Hom.isoRestrict : X ≅ Y.restrict H.base_open :=
```
will not enable dot notation

File: Mathlib/AlgebraicGeometry/AffineScheme.lean [hunk 2]
```diff
@@ -407,8 +407,8 @@ theorem exists_basicOpen_le {V : Opens X} (x : V) (h : ↑x ∈ U) :
     refine (Scheme.image_basicOpen (X.ofRestrict U.openEmbedding) r).trans ?_
     rw [← Scheme.basicOpen_res_eq _ _ (eqToHom U.openEmbedding_obj_top).op,
       ← comp_apply, ← CategoryTheory.Functor.map_comp, ← op_comp, eqToHom_trans, eqToHom_refl,
-      op_id, CategoryTheory.Functor.map_id, Scheme.Hom.invApp,
-      PresheafedSpace.IsOpenImmersion.ofRestrict_invApp]
+      op_id, CategoryTheory.Functor.map_id, Scheme.Hom.invApp]
+    erw [PresheafedSpace.IsOpenImmersion.ofRestrict_invApp]
```
Comment (jjaassoonn) [inline Mathlib/AlgebraicGeometry/AffineScheme.lean:411]: not sure why this got worse

**Comment by erdOne** [comment]
LGTM thanks!
maintainer merge

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**